Enhancements and Corrections to eDB360 section 4

davidkurtz opened the pull request:
Some of the CDB changes in sections 4a to 4e have been reversed because they are breaking the generation of the line charts.
Additional charts have been added to 4a to report the behaviour of the shared_pool, larger_pool, java_pool and streams_pool across each PDB when edb360 is run in the root container.

A number of changes and corrections have been made in section 4h
1. The calculation of the total time waited for the top 25 events has been corrected.  Previously the totals did not agree with ASH data.  Now the same formula as the per event queries has been used (with the CASE statement on wait_time_mili).  Hours waited is now shown to 2 decimal places.  Some column titles have been specifically set and formatted.
1. A new line chart has been added showing the average latency per snap of top 15 wait events over time.  The idea being that artefacts of interest are when latencies for some events scale differently to that for others.  This is where the lines cross.
1. The calculation of the latency trend has been changed.  Previously it was the average of the average latency for each snap accumulated over all the samples up to that point in the report.  However, this gave equal weight to all of the samples.  So a different latency at a quiet time had an undue influence on the trend.  Instead, the new calculation uses a weighted average, where the weight is the number of waits for each event.  This is done by summing all the wait time to date and dividing it by the sum of the number of weights to date.

davidkurtz committed:
Merge pull request #17 from sqldb360/unstable

Merge pull request #65 from davidkurtz/master

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

Corrections to section 4h
1. removed unnecessary wait_class_id from analytic partition by clause.  It is already grouped by event_id.
2 The trend is the cumulative averge of equally weighted averages.  I changed it to a cumulative average waited by number of events counted instead
3. removed group by instance, and sum per instance in final query joined to snapshots that will aggregate latency per RAC instance producing incorrect results.

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Merge branch 'master' of https://github.com/davidkurtz/sqldb360

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Add line chart of top 15 latency of time

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

Bugfix calculation of total wait time to get top 25 events.

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_4h_io_waits_top_trend.sql

davidkurtz committed:
Update edb360_0b_pre.sql

clear column setting for cpu_load_threshold after using it.

davidkurtz committed:
Update edb360_4a_sga_stats.sql

Corrections to CDB approach.  Move back to using DBA_HIST_SGASTAT, but reported individual pools per CDB so that charts work correctly.

davidkurtz committed:
CDB Corrections

Remove CDB columns from reports that produce line charts.  The extra columns are breaking the production of the googlechart.

davidkurtz commented on sql/edb360_0b_pre.sql:
```diff
@@ -962,6 +962,7 @@ COL name CLE;
 COL value CLE;
 COL display_value CLE;
 COL update_comment CLE;
+COL cpu_load_threshold CLEAR
```
Clear column setting to prevent it bleeding formatting into section 4

abelmacias commented:
The query on cdb_sql_plan_dir_objects raises a ora-7445 [kprcdt]. It works if the NOTES column is removed.

davidkurtz committed:
Update edb360_6c_ash_sql.sql

Remove con_id from 6c - it doesn't make sense to group SQL_ID per PDB when the rest of the column doesn't either

davidkurtz committed:
Update edb360_4a_sga_stats.sql

davidkurtz committed:
Update edb360_4a_sga_stats.sql

davidkurtz committed:
Update edb360_6c_ash_sql.sql

Remove con_id from group to surpress duplicate SQL_IDs from different PDBs.

davidkurtz committed:
Contain Database Fixes

Remove CON_ID from line charts but retain in other queries.

mauropagano commented:
Thanks @davidkurtz for all your hard work on this!
